web/src/realm_playground.ts

```diff
@@ -8,6 +8,7 @@ type RealmPlayground = {
     name: string;
     pygments_language: string;
     url_template: string;
+    playground_url?: string;
```

Comment:
I think this change is wrong.

Reply:
Ohhh...Got it!!!. Addressed in 1c9268f91921604bab7d0f6e4ac304285e25c892 .

Reply:
I think `RealmPlayground` should have a `playground_url` property...the loop actually adds `playground_url` property to 
 each element of `playground_info` array. At runtime `playground_info` array is iterated over and template : 
`<a href="{{this/playground_url}}" target="_blank" rel="noopener noreferrer" class="popover_playground_link">`
is replaced with actual `playground_url` property.

Reply:
I think `RealmPlayground` should have a `playground_url` property... From what I could understand, the loop that you mentioned actually adds `playground_url` property to each element(ie. the `playground` object) of `playground_info` array.
 At runtime `playground_info` array is iterated over and `playground_url` property is fed into the template.

Reply:
This change is still wrong. Ths code in this module:
```
                for (const playground of playground_info) {                                             
                    const url_template = url_template_lib.parse(playground.url_template);               
                    playground.playground_url = url_template.expand({code: extracted_code});            
                }                                                                                       
                const popover_target = $view_in_playground_button.find(                                 
                    ".playground-links-popover-container",                                              
                )[0];                                                                                   
                toggle_playground_links_popover(popover_target, playground_info);                       
```

Is doing something evil -- it's writing this `playground_url` field, which is specialized to the specific code block we're looking at, into the global playground object recording what playgrounds are configured.

Replace this commit with reworking how this code and `toggle_playground_links_popover` process things to avoid doing that -- which likely involves a new type or a new `Map<playground, resolved_playground_url> parameter passed into `toggle_playground_links_popover`.

Reply:
Creating a `Map` might be a little ineffecient as we'll have convert `Map` back into `array `of object ie. `playground_info` to feed into the ` render_playground_links_popover ` template. I think the above approach might be a lil better.

Reply:
Creating a `Map` is efficient and makes sense. Added a prep commit!.